src/include/dns-server/dialog-main.rb

```diff
@@ -2329,10 +2145,32 @@ def HandleExpertZonesPage(key, event)
       nil
     end
 
+    # Write settings dialog
+    # @return `abort if aborted and `next otherwise
+    def WriteDialog
+      Wizard.RestoreHelp(Ops.get_string(@HELPS, "write", ""))
+      ret = DnsServer.Write
+      if ret
+        @status_component.adjust_service
+        :next
+      else
+        if Popup.YesNo(_("Saving the configuration failed. Change the settings?"))
+          :back
+        else
+          :abort
```

Comment:
Well, I actually didn't change the message of the behaviour. This is mostly old code.